### PR TITLE
[Feature][Connector-V2] Support multi-table sink feature for HBase

### PR DESCRIPTION
**Pull request by BruceWong96**
### Purpose of this pull request

[Feature][HBase] Support multi-table sink feature https://github.com/apache/seatunnel/issues/5652

### How was this patch tested?

new e2e test

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by BruceWong96** [comment]
@Hisoka-X 
@hailin0 
Hi, PTAL.
Thanks.

**Comment by Hisoka-X** [comment]
Thanks @BruceWong96 for contribute this feature! Could you follow the guide to open github action on your fork repository?  https://github.com/apache/seatunnel/pull/7169/checks?check_run_id=27312810188

**Comment by BruceWong96** [comment]
> Thanks @BruceWong96 for contribute this feature! Could you follow the guide to open github action on your fork repository? https://github.com/apache/seatunnel/pull/7169/checks?check_run_id=27312810188

OK

**Comment by BruceWong96** [comment]
@Hisoka-X 
Up to now, Do I need to address JDK 11 compatibility issues ? 
Looking forward to your reply !

**Comment by Hisoka-X** [comment]
> @Hisoka-X Up to now, Do I need to address JDK 11 compatibility issues ? Looking forward to your reply !

Yes. Before we merge PR, we should make sure all test case passed.
image

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf]: How does it support writing to multiple tables?

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf]: Please add test cases to write to tables with different structures

**Review by hailin0 — COMMENTED** [review]
Please update docs

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf
Comment (BruceWong96) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf]: > How does it support writing to multiple tables?

Maybe I have a problem with my understanding. Is it many-to-many that we're going to end up with ?

Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf]: @BruceWong96 

You decide how to design this function.

e.g. multiple tables as input upstream:
```
source {
  FakeSource {
    tables_configs = [
      {
          schema {
            table = "table-11111"
            fields {
                id = string
                age = int
            }
          }
      },
      {
          schema {
            table = "table-2222"
            fields {
               f1 = boolean
               f2 = boolean
               f3 = tinyint
            }
          }
      }
    ]
  }
}
```

Comment (BruceWong96) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/resources/fake_to_hbase_with_multipletable.conf]: > @BruceWong96
> 
> You decide how to design this function.
> 
> e.g. multiple tables as input upstream:
> 
> ```
> source {
>   FakeSource {
>     tables_configs = [
>       {
>           schema {
>             table = "table-11111"
>             fields {
>                 id = string
>                 age = int
>             }
>           }
>       },
>       {
>           schema {
>             table = "table-2222"
>             fields {
>                f1 = boolean
>                f2 = boolean
>                f3 = tinyint
>             }
>           }
>       }
>     ]
>   }
> }
> ```

```
env {
  parallelism = 1
  job.mode = "BATCH"
}


source {
  FakeSource {
    tables_configs = [
      {
          schema {
            table = "table-11111"
            fields {
                id = string
                age = int
            }
          }
      },
      {
          schema {
            table = "table-2222"
            fields {
               f1 = boolean
               f2 = boolean
               f3 = tinyint
            }
          }
      }
    ]
  }
}

sink {
  Hbase {
    zookeeper_quorum = "hadoop001:2181,hadoop002:2181,hadoop003:2181"
    tables_configs = [
      {
        source_table_name = "table-11111"
        table = "hbase_table_1"
        rowkey_column = ["id"]
        family_name {
          id = "cf1"
          age = "cf1"
        }
      },
      {
        source_table_name = "table-2222"
        table = "hbase_table_2"
        rowkey_column = ["f1"]
        family_name {
          f1 = "cf2"
          f2 = "cf2"
          f3 = "cf2"
        }
      }
    ]
  }
}

```
Please check whether this configuration meets the requirements. If so, I will develop according to this configuration.
Thanks.

**Comment by hailin0** [comment]
@BruceWong96 I found a duplicate PR， here https://github.com/apache/seatunnel/pull/6969/, This is the correct way to code, you can use it as a reference for learning

@TaoZex @BruceWong96  Are you willing to collaborate with each other?

**Comment by BruceWong96** [comment]
> @BruceWong96 I found a duplicate PR， here #6969, This is the correct way to code, you can use it as a reference for learning
> 
> @TaoZex @BruceWong96 Are you willing to collaborate with each other?

OK , I'm going to learn about it. So you can close this PR. Thanks.

**Comment by TaoZex** [comment]
> > @BruceWong96 I found a duplicate PR， here #6969, This is the correct way to code, you can use it as a reference for learning
> > @TaoZex @BruceWong96 Are you willing to collaborate with each other?
> 
> OK , I'm going to learn about it. So you can close this PR. Thanks.

I close that pr, you can use it as a reference to try to implement it. Feel free to contact me if you have any questions.

**Comment by BruceWong96** [comment]
> > > @BruceWong96 I found a duplicate PR， here #6969, This is the correct way to code, you can use it as a reference for learning
> > > @TaoZex @BruceWong96 Are you willing to collaborate with each other?
> > 
> > 
> > OK , I'm going to learn about it. So you can close this PR. Thanks.
> 
> I close that pr, you can use it as a reference to try to implement it. Feel free to contact me if you have any questions.

Thanks a lot . I will commit the code at a later time.

**Comment by hailin0** [comment]
hi @BruceWong96
Is there any progress on this task?

**Comment by BruceWong96** [comment]
> hi @BruceWong96 Is there any progress on this task?

Yes, I am doing some local tests, sorry for the delay, I will update the code soon.

Thanks.

**Comment by BruceWong96** [comment]
@hailin0  PTAL.
Thanks.

**Comment by BruceWong96** [comment]
@Hisoka-X  PTAL.
Thanks.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: Spark supported now.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: Why removed this test case?

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java]: We supported this feature in https://github.com/apache/seatunnel/blob/570bbb39a0de9deed60e4089823b80bc59ec3263/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactoryContext.java#L45

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java
Comment (hailin0) [inline seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java]: remove this method

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java
Comment (BruceWong96) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: Fixed.

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java
Comment (BruceWong96) [reply seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java]: OK Fixed.

**Comment by BruceWong96** [comment]
@hailin0 PTAL.
Thanks.

File: seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-hbase/src/main/java/org/apache/seatunnel/connectors/seatunnel/hbase/sink/HbaseSinkFactory.java]: We already invoked this method in https://github.com/apache/seatunnel/blob/df210ea73d5b14b36b9637fc4d12bee09fa8decd/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/TableSinkFactoryContext.java#L51
So we shouldn't invoke it in connector. Please remove it.

File: docs/zh/connector-v2/sink/Hbase.md
Comment (Hisoka-X) [inline docs/zh/connector-v2/sink/Hbase.md]: Seem like the doc not up to date.

**Review by Hisoka-X — APPROVED** [review]
LGTM if ci passes. Thanks @BruceWong96 !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**
